### DISCUSSION TITLE
Error starting agent - failed to peek sni and alpns

### DISCUSSION THREAD

**Discussion by aarongodin** [narrowlink/narrowlink#31] [Q&A] [answered]
I have gone through the basic tutorial, skipping the client for now as I'm publishing a web server through following that guide. I have the gateway running and am attempting to start an agent that I've configured with a `publish` token. The gateway is receiving the connection but emits this error:

```
2023-08-24T17:02:21.650546Z  WARN narrowlink_gateway::service::wss: failed to peek sni and alpns
```

The agent treats this as the gateway hanging up the connection. Any ideas as to what this would point to that's misconfigured?

**Comment by SajjadPourali** [comment, marked as answer]
> 

This error indicates that the TLS server received a malformed TLS connection – a connection that either lacks an SNI extension or sends a non-TLS packet to the TLS server. It may not necessarily be the fault of the connecting agent to the gateway and could result from internet scanning services or similar factors.

However, if you use the `Wss` (`HTTPS+Websocket`) protocol to connect to the gateway, you need to use a domain name. I suspect that you intended to connect to the gateway using the `Wss` protocol but used an IP address in the agent's configuration field.

------

You can also set the `RUST_LOG=narrowlink_gateway=trace` environment variable in your gateway's terminal and run it again to obtain detailed logging information.

1. Could you please verify that you are using a domain name in the agent's configuration?
2. You can also share the logs with the mentioned variable or the configuration for further checks.

**Reply by aarongodin** [reply]
Thanks for the quick reply. I was indeed using an IP Address rather than a domain name as the host for the agent config. I've corrected that and have verified the gateway has registered the agent.

**Reply by aarongodin** [reply]
I was also able to get a webserver on my laptop running through the agent and available on my domain. Awesome!
